model_compression_toolkit/quantizers_infrastructure/keras/config_serialization.py

```diff
@@ -42,54 +38,44 @@ def transform_enum(v: Any):
         return v.value
     return v
 
-
-def config_serialization(quantization_config: BaseNodeQuantizationConfig):
+def config_serialization(quantization_config: BaseQuantizerConfig):
     """
-    This function change BaseNodeQuantizationConfig to a dictionary
+    This function change BaseQuantizerConfig to a dictionary
     Args:
-        quantization_config: A BaseNodeQuantizationConfig for serialization
+        quantization_config: A BaseQuantizerConfig for serialization
 
-    Returns: A config dictionary of BaseNodeQuantizationConfig
+    Returns: A config dictionary of BaseQuantizerConfig
 
     """
-    config_data = {k: transform_enum(v) for k, v in quantization_config.__dict__.items() if
-                   v is not isinstance(v, Callable)}
-    config_data[IS_WEIGHTS] = isinstance(quantization_config, NodeWeightsQuantizationConfig)
-    config_data[IS_ACTIVATIONS] = isinstance(quantization_config, NodeActivationQuantizationConfig)
+    config_data = {k: transform_enum(v) for k, v in quantization_config.__dict__.items()}
+    config_data[IS_WEIGHTS] = isinstance(quantization_config, TrainableQuantizerWeightsConfig)
+    config_data[IS_ACTIVATIONS] = isinstance(quantization_config, TrainableQuantizerActivationConfig)
     return config_data
 
 
-def config_deserialization(in_config: dict) -> BaseNodeQuantizationConfig:
+def config_deserialization(in_config: dict) -> BaseQuantizerConfig:
     """
-    This function change config dictionary to it BaseNodeQuantizationConfig.
+    This function change config dictionary to it BaseQuantizerConfig.
     Args:
-        in_config:  A config dictionary of BaseNodeQuantizationConfig
+        in_config:  A config dictionary of BaseQuantizerConfig
 
-    Returns: A BaseNodeQuantizationConfig
+    Returns: A BaseQuantizerConfig
 
     """
     in_config = copy.deepcopy(in_config)
-    qc = QuantizationConfig()
-    op_cfg = OpQuantizationConfig(QuantizationMethod.POWER_OF_TWO, QuantizationMethod.POWER_OF_TWO,
-                                  8, 8, True, True, True, True, 0, 0, 8)
     if in_config[IS_WEIGHTS]:
-        nwqc = NodeWeightsQuantizationConfig(qc=qc,
-                                             op_cfg=op_cfg,
-                                             weights_quantization_fn=None,
-                                             weights_quantization_params_fn=None,
-                                             weights_channels_axis=0)
-        in_config[WEIGHTS_QUANTIZATION_METHOD] = QuantizationMethod(in_config[WEIGHTS_QUANTIZATION_METHOD])
-
-        nwqc.__dict__.update(in_config)
-        return nwqc
+        return TrainableQuantizerWeightsConfig(weights_quantization_method=QuantizationMethod(in_config['weights_quantization_method']),
```

Comment:
Why not use the constant variable? (and put it in "constants" file)

Reply:
note that it's a field name of the config class, so you'll need to use set it in the class __init__ with setattr(CONST, value) instead of self.CONST = value